src/views/Question.vue

```diff
@@ -4,12 +4,18 @@
       <div class="question-category">{{ question.title }}</div>
       <div class="question-text">{{ question.questionText }}</div>
         <div v-if="question.type === 'Single choice'">
-          <QuizSingleChoice :choices="question.answers"></QuizSingleChoice>
+          <QuizSingleChoice :choices="question.answers" @answering="select"></QuizSingleChoice>
         </div>
         <div v-else-if="question.type === 'Multiple choice'">
           <QuizMultiChoice :choices="question.answers"></QuizMultiChoice>
         </div> 
-      <button class='button-disabled'><v-icon name="arrow-right"/></button>
+
+        <router-link :to="{ name: 'question', params: { category: this.$route.params.category, id: nextQuestionLink}}" :class="{'router-disabled': inputSelected.length == 0}">
+          <button :class="buttonState" @click="submit">
```

Comment:
Somehow both `router-link` and `button` here are intercepting the click. Also, `buttonState` seems to refer to `inputSelected.length` state as well (both actually works the same way).

It's better to combine both the linking and the `submit` function together, by removing `router-link` component and using [router.push](https://router.vuejs.org/api/#router-push) instead inside the `submit` function.